lib/reporting/report_headers_builder.rb

```diff
@@ -18,12 +18,15 @@ def available_headers
       report.columns.keys.map { |key| [translate_header(key), key] }
     end
 
-    def fields_to_hide
-      if report.display_header_row?
-        report.formatted_rules.map { |rule| rule[:fields_used_in_header] }.flatten.reject(&:blank?)
-      else
-        []
-      end.concat(params_fields_to_hide)
+    def fields_to_show
+      fields_in_headers = if report.display_header_row?
+                            report.formatted_rules.map { |rule|
+                              rule[:fields_used_in_header]
+                            }.flatten.reject(&:blank?)
+                          else
+                            []
+                          end
+      params_fields_to_show - fields_in_headers
```

Comment:
Without reading all the code, I don't understand this statement. Why do subtract the header fields from the fields to show? Or why do we *not* show header fields?

Reply:
I'll talk under the control of @seballot but as I can see, `fields_used_in_header` are fields that:
> will be hidden when the header_row is activated

So we need to remove them in order to be shown (don't know if I'm clear)

Reply:
Yes that's it. If the table have given_name / family_name / product / price, and we activate the header_row and the header_row display given_name && family_name, then we only want to display product and price column
```
GivenName | FamilyName | Product1 | XX €
GivenName | FamilyName | Product2 | XX €

Turns into

Header: Given_name Family Name
Product1 | XX €
Product2 | XX €
```

Reply:
Ah, thank you for that example. I wasn't aware of that transformation. Interesting. We can probably find a better way to represent that but that's out of scope here.